lib/ascii_binder/engine.rb

```diff
@@ -0,0 +1,872 @@
+
+      if root_group
+        root_topic = root_group[:topics].first
+        breadcrumb_root = linkify_breadcrumb(root_topic[:path], "#{args[:distro]} #{args[:version]}", current_is_subtopic) if root_topic
+      end
+
+      if selected_group
+        group_topic = selected_group[:topics].first
+        breadcrumb_group = linkify_breadcrumb(group_topic[:path], selected_group[:name], current_is_subtopic) if group_topic
+
+        if selected_subgroup
+          subgroup_topic = selected_subgroup[:topics].first
+          breadcrumb_subgroup = linkify_breadcrumb(subgroup_topic[:path], selected_subgroup[:name], current_is_subtopic) if subgroup_topic
+
+          selected_topic = selected_subgroup[:topics].detect { |topic| topic[:id] == args[:topic_id] }
+          breadcrumb_topic = linkify_breadcrumb(nil, selected_topic[:name], current_is_subtopic) if selected_topic
+        else
+          selected_topic = selected_group[:topics].detect { |topic| topic[:id] == args[:topic_id] }
+          breadcrumb_topic = linkify_breadcrumb(nil, selected_topic[:name], current_is_subtopic) if selected_topic
+        end
+      end
+
+      return breadcrumb_root, breadcrumb_group, breadcrumb_subgroup, breadcrumb_topic
+    end
+
+    def linkify_breadcrumb(href, text, extra_level)
+      addl_level = extra_level ? '../' : ''
+      href ? "<a href=\"#{addl_level}#{href}\">#{text}</a>" : text
+    end
+
+    def parse_distros distros_string, for_validation=false
+      values   = distros_string.split(',').map(&:strip)
+      # Don't bother with glob expansion if 'all' is in the list.
+      return distro_map.distro_keys if values.include?('all')
+
+      expanded = expand_distro_globs(values)
+      return expanded if for_validation
+      return expanded.uniq
+    end
+
+    def expand_distro_globs(values)
+      values.flat_map do |value|
+        value_regex = Regexp.new("\\A#{value.gsub("*", ".*")}\\z")
+        distro_map.distro_keys.select { |k| value_regex.match(k) }
+      end.uniq
+    end
+
+    def validate_distros distros_string
+      return false if not distros_string.is_a?(String)
+      values = parse_distros(distros_string, true)
+      values.each do |v|
+        return false if not v == 'all' and not distro_map.distro_keys.include?(v)
+      end
+      return true
+    end
+
+    def validate_topic_group group, info
+      # Check for presence of topic group keys
+      ['Name','Dir','Topics'].each do |group_key|
+        if not group.has_key?(group_key)
+          raise "One of the topic groups in #{build_config_file} is missing the '#{group_key}' key."
+        end
+      end
+      # Check for right format of topic group values
+      ['Name','Dir'].each do |group_key|
+        if [true, false].include?(group[group_key])
+          raise "One of the topic groups in #{build_config_file} is using a reserved YAML keyword for the #{group_key} setting. In order to prevent your text from being turned into a true/false value, wrap it in quotes."
```

Comment:
In general, changes to the parsing for `_topic_map.yml` (the artist formerly known as `_build_cfg.yml`) are all slated for the next refactor PR.

For this specific question - it doesn't look like [YAML](https://ruby-doc.org/stdlib-2.4.1/libdoc/yaml/rdoc/YAML.html) or [Psych](http://www.rubydoc.info/gems/psych/2.0.17/Psych) preserve line numbers, but I can probably provide a better breadcrumb trail for tracking down issues.